plugins/gatsby-plugin-auto-index-pages/gatsby-node.js

```diff
@@ -0,0 +1,125 @@
+const path = require('path');
+const fromList = require('./utils/unist-fs-util-from-list');
+const visit = require('unist-util-visit');
+const generateHTML = require('./utils/generate-html');
+const { sentenceCase } = require('./utils/string');
+
+exports.createPages = async ({ actions, graphql, reporter }, pluginOptions) => {
+  const { skippedDirectories } = pluginOptions;
+  const { createPage } = actions;
+
+  const { data, errors } = await graphql(`
+    query MyQuery {
+      tableOfContents: allFile(
+        filter: {
+          sourceInstanceName: { eq: "markdown-pages" }
+          base: { in: ["index.mdx", "index.md"] }
+        }
+      ) {
+        nodes {
+          childMdx {
+            fields {
+              slug
+            }
+          }
+          childMarkdownRemark {
+            fields {
+              slug
+            }
+          }
+        }
+      }
+      allFile(
+        sort: { fields: [relativePath] }
+        filter: {
+          sourceInstanceName: { eq: "markdown-pages" }
+          base: { nin: ["index.mdx", "index.md"] }
+          children: {
+            elemMatch: { internal: { type: { in: ["MarkdownRemark", "Mdx"] } } }
+          }
+        }
+      ) {
+        nodes {
+          base
+          relativePath
+          childMdx {
+            frontmatter {
+              title
+            }
+            fields {
+              slug
+            }
+          }
+          childMarkdownRemark {
+            frontmatter {
+              title
+            }
+            fields {
+              slug
+            }
+          }
+        }
+      }
+    }
+  `);
+
+  if (errors) {
+    reporter.panicOnBuild(errors[0]);
+  }
+
+  const {
+    tableOfContents: { nodes: tableOfContentsNodes },
+    allFile: { nodes: fileNodes },
+  } = data;
+
+  const existingPaths = tableOfContentsNodes
+    .map(getSlug)
+    .concat(fileNodes.map(getSlug));
+
+  const files = fileNodes.map((node) => ({
+    basename: node.base,
+    path: node.relativePath,
+    contents: node.childMdx || node.childMarkdownRemark,
+  }));
+
+  const list = fromList(files, ({ contents }) => contents);
+
+  visit(list, 'directory', (dir) => {
+    const slug = `/${dir.path}`;
+
+    if (skippedDirectories.includes(dir.path)) {
+      return [visit.SKIP];
+    }
+
+    if (existingPaths.includes(slug)) {
+      return;
+    }
+
+    createPage({
```

Comment:
so this is creating an index page, and the path below is creating a table of contents page, what is the difference between these two? (sadly wish i could build to visualize 😢 )

Reply:
They are similar, but the table of contents _pages_ are the pages that already have a landing page associated with them. For example, here is a landing page: https://docs-preview.newrelic.com/docs/agents/java-agent. And here is its associated table of contents page (which you can navigate to by clicking the button underneath the tiles): https://docs-preview.newrelic.com/docs/agents/java-agent/table-of-contents

The table of contents page is structured based on the nav, whereas the index pages are structured based on the child URLs in any given route. They look visually similar though.

Reply:
gotcha! thanks for the context